src/site/content/en/blog/enabling-shared-array-buffer/index.md

```diff
@@ -0,0 +1,92 @@
+  - api
+---
+
+It's fair to say `SharedArrayBuffer` has had a bit of a rough landing on the web, but things are settling down. Here's what you need to know:
+
+## In brief
+
+- `SharedArrayBuffer` will arrive in Android Chrome 88, but only for cross-origin isolated pages (details below).
+- `SharedArrayBuffer` is currently available in desktop Chrome, but from Chrome 91 it will be limited to cross-origin isolated pages.
+
+You can make a page _cross-origin isolated_ by serving the page with these headers:
+
+```http
+Cross-Origin-Embedder-Policy: require-corp
+Cross-Origin-Opener-Policy: same-origin
+```
+
+Once you do this, your page will not be able to load cross-origin content unless the [`Cross-Origin-Resource-Policy` header](<https://developer.mozilla.org/en-US/docs/Web/HTTP/Cross-Origin_Resource_Policy_(CORP)>) allows it, or if it's requested using [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS), and the usual `Access-Control-Allow-*` headers allow it.
+
+If you don't think you can make these changes in time for Chrome 91, you can [register for an origin trial](#register-for-ot) to retain current desktop Chrome behavior until Chrome 93.
+
+More info on:
+
+- [How to cross-origin isolate your pages](/coop-coep/).
+- [Why cross-origin isolation is needed](/why-coop-coep/).
+
+## How did we get here?
+
+`SharedArrayBuffer` arrived in Chrome 60 (that's July 2017, for those of you who think of time in dates rather than Chrome versions), and everything was great. For 6 months.
+
+In January 2018 a vulnerability was revealed in some popular CPUs. [See the announcement](https://googleprojectzero.blogspot.com/2018/01/reading-privileged-memory-with-side.html) for full details, but it essentially meant that code could use high resolution timers to read memory that it shouldn't have access to.
+
+This was a problem for us browser vendors, as we want to allow sites to execute code in the form of JavaScript and WASM, but strictly control the memory this code can access. If you arrive on my website, I shouldn't be able to read anything from the internet banking site you also have open. In fact, I shouldn't even know you have your internet banking site open. These are fundamentals of web security.
+
+To mitigate this, we reduced the resolution of our high resolution timers such as `performance.now()`. However, you can _create_ a high resolution timer using `SharedArrayBuffer` by modifying memory in a tight loop in a worker, and reading it back in another thread. This couldn't be effectively mitigated without heavily impacting well-intentioned code, so `SharedArrayBuffer` was disabled altogether.
+
+A general mitigation is to ensure a webpage's system process doesn't contain sensitive data from elsewhere. Chrome had invested in a mutliprocess architecture from the start ([remember the comic?](https://www.google.com/googlebooks/chrome/big_00.html)), but there were still cases where data from multiple sites could end up in the same process:
```

Comment:
```suggestion
A general mitigation is to ensure a webpage's system process doesn't contain sensitive data from elsewhere. Chrome had invested in a multiprocess architecture from the start ([remember the comic?](https://www.google.com/googlebooks/chrome/big_00.html)), but there were still cases where data from multiple sites could end up in the same process:
```

Reply:
lol even with the diff it took me ages to see the error. Thanks!

![0mutlí](https://user-images.githubusercontent.com/93594/100982901-888c3080-3540-11eb-87be-9d3f439eca6d.png)